### ISSUE TITLE
Deleting a resource from the edit page with the UsesResourceLock trait results in an error before redirecting

### ISSUE THREAD

**Issue by adriaardila** [kenepa/resource-lock#39]
When attempting to delete a resource from the edit page, it throws an error before redirecting the user to the index page / table.

![image](https://github.com/kenepa/resource-lock/assets/2029010/bdc5fc08-940f-4a37-91be-6a05e0bab6ec)

It happens on all edit pages (for example EditContact) that have the use usesresourcelock trait

**Comment by Jehizkia** [comment]
Hi, thank you for reporting. Could you please share the entire error page of your screenshot with the share button on the top right so I can inspect it?

**Comment by adriaardila** [comment]
> Hi, thank you for reporting. Could you please share the entire error page of your screenshot with the share button on the top right so I can inspect it?

Here you go:

https://flareapp.io/share/NPLojdgm

I don't think the error will be of much help since it does not crash on your code specifically, the problem is that when you delete a model the $record ceases to exist, so maybe in the trait you need to check that the $record is initialized before attempting to access it

**Comment by Jehizkia** [comment]
Looking at the errors, it fails when the `getRecord()` function is called. But the `UsesResourceLock` trait doesn't call this function at all. Seems like this originates from the `RelationManager`.

**Comment by adriaardila** [comment]
@Jehizkia yes, but once I remove the HasLocks trait it works fine, the HasLocks trait uses the $record variable, the problem is that it doesn't check that that variable is always existant and when you delete a record there's a split second where the Livewire component of the EditRecord is reloaded, and then attempts to load the $record called in your trait but that record does not exist anymore. Maybe could be worth checking that the $record is initialized before attempting to access it in the trait.

I can do a PR later and see if I can fix it.

Thank you

**Comment by Jehizkia** [comment]
@adriaardila Ah, I see what you mean. Yeah, if you could whip something up, I’ll give it a look.👌

**Comment by adriaardila** [comment]
@Jehizkia sure, btw I also added something that I need for my project and did a PR on that, it's an option to disable the navigation on the sidebar if needed.

https://github.com/kenepa/resource-lock/pull/41
